src/Octoshift/HttpDownloadServiceFactory.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using OctoshiftCLI.Contracts;
+
+namespace OctoshiftCLI.GithubEnterpriseImporter
+{
+    public sealed class HttpDownloadServiceFactory : IHttpDownloadServiceFactory
+    {
+        private readonly OctoLogger _log;
+        private readonly IHttpClientFactory _clientFactory;
+        private readonly FileSystemProvider _fileSystemProvider;
+        private readonly IVersionProvider _versionProvider;
+
+        public HttpDownloadServiceFactory(OctoLogger log, IHttpClientFactory clientFactory, FileSystemProvider fileSystemProvider, IVersionProvider versionProvider)
+        {
+            _log = log;
+            _clientFactory = clientFactory;
+            _fileSystemProvider = fileSystemProvider;
+            _versionProvider = versionProvider;
+        }
+
+        public HttpDownloadService Create()
+        {
+            var httpClient = _clientFactory.CreateClient();
+            ConfigureClient(httpClient);
+
+            return new HttpDownloadService(_log, httpClient, _fileSystemProvider);
+        }
+
+        public HttpDownloadService CreateDefault()
+        {
+            var httpClient = _clientFactory.CreateClient("Default");
+            ConfigureClient(httpClient);
+
+            return new HttpDownloadService(_log, httpClient, _fileSystemProvider);
+        }
+
+        public HttpDownloadService CreateClientNoSsl()
+        {
+            var httpClient = _clientFactory.CreateClient("NoSSL");
+            ConfigureClient(httpClient);
+
+            return new HttpDownloadService(_log, httpClient, _fileSystemProvider);
+        }
```

Comment:
For simplicity and ease of use I would create just a single `Create` method with a boolean `noSsl` flag that is set t `false` by default and then create the client based of the flag --> `Create(noSsl = false)`